### PR TITLE
Support full set of pipeline query objects options

### PR DESCRIPTION
**Pull request by nahiluhmot**
@tucker250 

Instead of modifying `#query_objects` to strictly pull all of the queried objects, I added a new method, `#query_all_objects` that automatically follows the pagination. I thought that this may be better in case somebody would like to lazily pull the objects in their own library.

### CODE REVIEW

File: lib/fog/aws/requests/data_pipeline/query_objects.rb [hunk 1]
```diff
@@ -27,10 +40,48 @@ def query_objects(id, sphere)
...
+        #         * 'Type'<~String>
+        #         * 'Values'<~Array<String>>
+        # ==== Returns
+        # * response<~Excon::Response>:
+        #   * 'Ids'<~Array<String>> - A list of identifiers that match the query selectors.
+        def query_all_objects(id, sphere, options={})
+          has_more_results = true
+          objects = []
+
+          while has_more_results
```
Comment (tucker250) [inline lib/fog/aws/requests/data_pipeline/query_objects.rb]: I like this idea, but can you check and see how pagination is handled elsewhere in fog first?  I'm not sure whether it has convenience methods to handle pagination for the other API calls that support it.

**Comment by mattgillooly** [comment]
Should we be opening pull requests against the master branch of our fork like this?  My usual workflow is to open the pull from my topic branch to fog/fog master.  @kbarrette

**Comment by tucker250** [comment]
Tom asked that as well and I recommended this approach.  I thought that we wanted to work on our local fork so we don't have to wait on fog's release cycle to start using our changes.

**Comment by mattgillooly** [comment]
So, is the idea that all our projects always point at our fork instead of getting the gem from rubygems.org?

**Comment by mattgillooly** [comment]
Sorry for the accidental close.

**Comment by kbarrette** [comment]
If we're going to go this route, we need to be responsible about keeping this up to date.

**Comment by tucker250** [comment]
That seems ok to me, we're doing something similar with delayed_job_active_record.  The only alternative I can think of is pointing our Gemfile at the branch on fog's github page, but that's fragile because someone else could delete the branch and break our build.  I think it's fair to point to our own fork whenever we use a gem we're contributing to.

**Comment by kbarrette** [comment]
Just playing devil's advocate here, but:

delayed_job_active_record is a different case.  We've acknowledged that what we're doing there is unlikely to ever get merged into the main line without a bunch more work on our part - work that we're not ready to do right now.  This  IS meant to be merged.

I'm ok with maintaining our own fork, but I really want to make sure that we keep our fork up to date, AND that our work gets merged into the main fog repo.

**Comment by tucker250** [comment]
@nahiluhmot added a checkbox to the Trello card to ensure we don't forget about sending a pull to fog/fog master: https://trello.com/c/PahykyfI  Then, we'll point our Gemfile to our fork's master branch until there's a fog release with our changes.

@mattgillooly @kbarrette - good enough solution?

**Comment by nahiluhmot** [comment]
@tucker250 I removed the pagination code -- good to merge to `swipely/fog/master`?

**Comment by tucker250** [comment]
@nahiluhmot - As soon as it's working in the console.

**Comment by nahiluhmot** [comment]
Validated on the console. @tucker250 Add a `.sw.1` to the version and merge?

**Comment by tucker250** [comment]
@nahiluhmot - Sounds good!

**Pull request merged**